### PR TITLE
allow + in database names during setup

### PR DESCRIPTION
**Pull request by nbezzala**
Fix for #9844

### CODE REVIEW

**Comment by tsteur** [comment]
Looks good to me. Thx for the PR 👍

File: plugins/Installation/FormDatabaseSetup.php [hunk 1]
```diff
@@ -302,7 +303,7 @@ private function dropExtraTables($db)
 }
 
 /**
- * Filename check for prefix/DB name
+ * Filename check for prefix
  *
  */
 class FormDatabaseSetup_Rule_checkValidFilename extends HTML_QuickForm2_Rule
```
Comment (mattab) [inline plugins/Installation/FormDatabaseSetup.php]: maybe table prefix also can't contain `+`, if so:  should we also validate the table prefix with the `DbHelper::isValidDbname`?

**Comment by nbezzala** [comment]
Yes, table names can contain a `+` in then. So, we should validate with DbHelper::isValidDbName. 
I wrote a patch to do that, and I changed the Mysql schema file to contain all table names inside ` characters, 
but it is failing, and I don't see why it is failing.

**Comment by nbezzala** [comment]
I tried changing the Common::prefixTable to put the table names within ` characters. But this is causing some other code to fail because they already are using the ` character around their table names.
For example - 
tests/PHPUnit/Integration/OptionTest.php:27:        Db::query("INSERT INTO `" . Common::prefixTable('option') . "` VALUES ('anonymous_defaultReport', '1', false)");

We need some kind of convention for this which everyone will use. And once the decision is made, we can make the change in all the places where prefixTable is called. 

I think it will take some time, so you need to decide if it is worth doing.

**Comment by nbezzala** [comment]
mysql> select * from ab+cduser;
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '+cduser' at line 1

mysql> select * from `ab+cduser`;
+-----------+----------+-----------+-----------------------+------------+------------------+---------------------+
| login     | password | alias     | email                 | token_auth | superuser_access | date_registered     |
+-----------+----------+-----------+-----------------------+------------+------------------+---------------------+
| anonymous |          | anonymous | anonymous@example.org | anonymous  |                0 | 2016-04-29 21:41:05 |
+-----------+----------+-----------+-----------------------+------------+------------------+---------------------+
1 row in set (0.00 sec)

mysql> select * from ``ab+cduser``;
ERROR 1103 (42000): Incorrect table name ''

**Pull request merged**